### DISCUSSION TITLE
Where did SimplePointInRing go?

### DISCUSSION THREAD

**Discussion by jdege** [NetTopologySuite/NetTopologySuite#510] [Q&A] [answered]
We're moving a project from .NET 4.5.2 to .NET 5.0, and from NetTopologySuite v1.14.0 to v2.2.0.

We have a number of places where we're using NetTopologySuite.Algorithm.SimplePointInRing(), but that seems to no longer exist.

Has it been deprecated? Removed? Moved to a different NuGet package?

Or is there some other mechanism for determining whether a point is inside a ring, that is preferred?

**Comment by airbreather** [comment, marked as answer]
[`NetTopologySuite.Algorithm.PointLocation.IsInRing`](http://nettopologysuite.github.io/NetTopologySuite/api/NetTopologySuite.Algorithm.PointLocation.html#NetTopologySuite_Algorithm_PointLocation_IsInRing_NetTopologySuite_Geometries_Coordinate_NetTopologySuite_Geometries_Coordinate___) (or an overload) is good if each ring is tested against very few points.

If you have many points to test against each ring, I recommend making an instance of [`NetTopologySuite.Algorithm.Locate.IndexedPointInAreaLocator`](http://nettopologysuite.github.io/NetTopologySuite/api/NetTopologySuite.Algorithm.Locate.IndexedPointInAreaLocator.html) for each ring and testing the points against that.

**Reply by jdege** [reply]
Thanks. That may work.
(Don't really know if it will, yet, because the lazy bastards who wrote the code I'm porting didn't think to create any unit tests, but that's a different problem...)
